### PR TITLE
chore: depend on public ethereum_ssz and ethereum_serde_utils crates

### PR DESCRIPTION
**Pull request by morph-dev**
### What was wrong?

We had dependency on private fork of `ethereum_ssz` and `ethereum_serde_utils`.

### How was it fixed?

1. Adding `ssz` feature to `alloy-primitives` eliminated need for custom version of `ethereum_ssz`.

2. The custom function `u256_from_dec_str` from `ethereum_serde_utils` that Kolby added wasn't really needed because `alloy-primitives` has `U256::from_str_radix(...)`.

3. In order for everything to work, I also needed to fork @KolbyML 's `ssz_types` repo and make it depend on public `ethereum_ssz` and `ethereum_serde_utils` as well (commit [link](https://github.com/morph-dev/kolby_ssz_types/commit/e6129bf860cea549a510281222a13154f3370ac1)).
    - Going forward, we can merge my changes into his repo, as he is more likely to keep maintaining it long term.

This leaves us with only 3 non-public repos:
- [morph-dev/kolby_ssz_types](https://github.com/morph-dev/kolby_ssz_types/tree/alloy) - only trivial changes, we can merge them into @KolbyML 's repo
- [KolbyML/tree_hash](https://github.com/KolbyML/tree_hash/tree/alloy) - relatively simple changes
- [KolbyML/eth-trie.rs](https://github.com/KolbyML/eth-trie.rs/tree/trin2-alloy) - we can leave it for now as it is and evaluate as we progress with more work on state network

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history and use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

### CODE REVIEW

**Review by njgheorghita — APPROVED** [review]
Nice! Looks good to me. Agreed that we should merge your `ssz_types` changes into @KolbyML 's repository. Whether or not that happens before or after merging this pr, I'll leave up to @KolbyML

**Comment by KolbyML** [comment]
https://github.com/KolbyML/ssz_types/pull/1 I just merged this PR, cool find

**Review by KolbyML — APPROVED** [review]
:shipit:

**Pull request merged**
